### PR TITLE
[image-builder] Backport mkIII from gitpod-com

### LINKED ISSUE #4812: Merge image-builder mkIII
The image-builder mkIII PR (#4547) has been sitting around for far too long. Let's get this merged (solve https://github.com/gitpod-io/gitpod/issues/4804 in the process) and make it available for 20% of image builds.

Once it's proven stable, we can remove the old image builder.

## Steps
1. rebase #4547
2. integrate with latest headless log infrastructure

### PR DESCRIPTION
**Pull request by csweichel**
This PR backports the image-builder mkIII from the gitpod-com repo. image-builder mkIII is the third rewrite of image-builder and works fundamentally different to is predecessors: 
- all image builds run as headless workspaces, which means that image-builds scale much nicer compared to mkII
- no more Docker-in-Docker, but uses buildkit directly which makes for a simpler implementation and lighter dependencies
- uses containerd's fetcher to resolve images

While its implementation is really different to previous iterations it maintains the same interface and is a drop-in replacement for image-builder mkII.

Note: because this is essentially a backport of previous work it comes in a single commit.

### ToDo
- [x] Make sure this stuff even builds
- [x] Test image builds and make sure they even still work
- [x] Sanity check the implementation - so far I've just blindly copied over what I had written more than a year ago
- [x] Integrate feedback from https://github.com/gitpod-io/gitpod/pull/4251

### Future Work
- ws-manager selection is currently fixed in the configuration. This does not sit well with pluggable workspace cluster. Instead we should select the target workspace cluster by asking ws-manager-bridge (as should server). For now we'll keep running the image builds in the fat cluster.

### Known defects
- [x] image builder mkIII uses the buildID as workspace instance ID, which means it might re-use the same instance ID if the previous build failed. ws-manager and ws-daemon don't cope too well with re-used instance ID's, e.g. the `initializerMap` in `monitor.go`

This PR supersedes https://github.com/gitpod-io/gitpod/pull/4251 and installs image-builder mk3 alongside the previous image builder to facilitate a smoother transition.

fixes #4812

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/4547#" title="Author self-approved">csweichel</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **jankeromnes** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @jankeromnes` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)~~ [csweichel]
- ~~[components/image-builder/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/image-builder/OWNERS)~~ [csweichel]
- ~~[components/image-builder-api/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/image-builder-api/OWNERS)~~ [csweichel]
- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**
- ~~[components/ws-manager/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-manager/OWNERS)~~ [csweichel]
- ~~[components/ws-manager-api/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-manager-api/OWNERS)~~ [csweichel]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/4547#" title="Author self-approved">csweichel</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **geropl** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @geropl` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)~~ [csweichel]
- **[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)**
- **[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)**
- ~~[components/image-builder/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/image-builder/OWNERS)~~ [csweichel]
- ~~[components/image-builder-api/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/image-builder-api/OWNERS)~~ [csweichel]
- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**
- ~~[components/ws-manager/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-manager/OWNERS)~~ [csweichel]
- ~~[components/ws-manager-api/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-manager-api/OWNERS)~~ [csweichel]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/4547#" title="Author self-approved">csweichel</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **geropl** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @geropl` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)~~ [csweichel]
- **[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)**
- **[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)**
- ~~[components/image-builder/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/image-builder/OWNERS)~~ [csweichel]
- ~~[components/image-builder-api/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/image-builder-api/OWNERS)~~ [csweichel]
- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**
- ~~[components/ws-manager/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-manager/OWNERS)~~ [csweichel]
- ~~[components/ws-manager-api/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-manager-api/OWNERS)~~ [csweichel]
- ~~[dev/gpctl/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/dev/gpctl/OWNERS)~~ [csweichel]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/4547#" title="Author self-approved">csweichel</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **jankeromnes** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @jankeromnes` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)~~ [csweichel]
- ~~[components/image-builder/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/image-builder/OWNERS)~~ [csweichel]
- ~~[components/image-builder-api/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/image-builder-api/OWNERS)~~ [csweichel]
- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**
- ~~[components/ws-manager/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-manager/OWNERS)~~ [csweichel]
- ~~[components/ws-manager-api/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-manager-api/OWNERS)~~ [csweichel]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/4547#" title="Author self-approved">csweichel</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **geropl** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @geropl` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)~~ [csweichel]
- ~~[components/image-builder/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/image-builder/OWNERS)~~ [csweichel]
- ~~[components/image-builder-api/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/image-builder-api/OWNERS)~~ [csweichel]
- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**
- ~~[components/ws-manager/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-manager/OWNERS)~~ [csweichel]
- ~~[components/ws-manager-api/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-manager-api/OWNERS)~~ [csweichel]
- ~~[dev/gpctl/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/dev/gpctl/OWNERS)~~ [csweichel]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: components/image-builder-bob/go.mod [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+	github.com/containerd/containerd v1.5.2
+	github.com/docker/docker v20.10.6+incompatible // indirect
+	github.com/gitpod-io/gitpod/common-go v0.0.0-00010101000000-000000000000
+	github.com/gofrs/flock v0.8.0 // indirect
+	github.com/gogo/googleapis v1.4.1 // indirect
+	github.com/golang/groupcache v0.0.0-20210331224755-41bb18bfe9da // indirect
+	github.com/golang/protobuf v1.5.2 // indirect
+	github.com/grpc-ecosystem/go-grpc-middleware v1.3.0 // indirect
+	github.com/jaguilar/vt100 v0.0.0-20201024211400-81de19cb81a4 // indirect
+	github.com/moby/buildkit v0.8.3
```
Comment (aledbf) [inline components/image-builder-bob/go.mod:18]: Please use 0.9.0

**Comment by aledbf** [comment]
I run several examples, some started the build and in one (`gitpod-io`) I got this error:
`Error: container workspace completed; containers of a workspace pod are not supposed to do that`

**Comment by svenefftinge** [comment]
I was testing this, by starting a workspace that requires a docker build (https://cw-imgbuilder-mk3-rebase.staging.gitpod-dev.com/#https://github.com/gitpod-io/template-haskell)
The log stood empty for a minute before it started to show `connecting to log output ...`. 
After another minute or so, all logs appeared at once. 
The docker build fails so it is reproducible.

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/4547#" title="Author self-approved">csweichel</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **jankeromnes** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @jankeromnes` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)~~ [csweichel]
- ~~[components/image-builder/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/image-builder/OWNERS)~~ [csweichel]
- ~~[components/image-builder-api/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/image-builder-api/OWNERS)~~ [csweichel]
- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**
- ~~[components/ws-manager/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-manager/OWNERS)~~ [csweichel]
- ~~[components/ws-manager-api/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-manager-api/OWNERS)~~ [csweichel]
- ~~[components/ws-proxy/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-proxy/OWNERS)~~ [csweichel]
- ~~[dev/gpctl/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/dev/gpctl/OWNERS)~~ [csweichel]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by csweichel** [comment]
Uploading Screen Recording 2021-07-20 at 21.53.29.mov…

The log output is much smoother now - albeit not yet optimal (i.e. it's just not very pretty)

File: components/image-builder-bob/entrypoint.sh [hunk 2]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+
+echo '{"message":{"type": "workspaceTaskDone"}}'
```
Comment (geropl) [inline components/image-builder-bob/entrypoint.sh:3]: This is a leftover, right?

File: components/ws-manager/pkg/manager/config.go [hunk 3]
```diff
@@ -132,6 +132,8 @@ type WorkspacePodTemplateConfiguration struct {
 	ProbePath string `json:"probePath,omitempty"`
 	// GhostPath is a path to an additional workspace pod template YAML file for ghost workspaces
 	GhostPath string `json:"ghostPath,omitempty"`
+	// ImagebuildPath is a oath to an additional workspace pod template YAML file for imagebuild workspaces
```
Comment (geropl) [inline components/ws-manager/pkg/manager/config.go:135]: typo: `oath` -> `path`

**Comment by geropl** [comment]
When trying to open https://cw-imgbuilder-mk3-rebase.staging.gitpod-dev.com/#https://github.com/geropl/roborally I get this error:
```

#5 [2/6] RUN bash -lc "npm install -g gatsby-cli"
#5 0.258 container_linux.go:380: starting container process caused: process_linux.go:545: container init caused: rootfs_linux.go:76: mounting "proc" to rootfs at "/proc" caused: mount through procfd: operation not permitted
#5 ERROR: process "/bin/sh -c bash -lc \"npm install -g gatsby-cli\"" did not complete successfully: exit code: 1
------
 > [2/6] RUN bash -lc "npm install -g gatsby-cli":
#5 0.258 container_linux.go:380: starting container process caused: process_linux.go:545: container init caused: rootfs_linux.go:76: mounting "proc" to rootfs at "/proc" caused: mount through procfd: operation not permitted
------
FATA[0153] build failed                                  error="failed to solve: process \"/bin/sh -c bash -lc \\\"npm install -g gatsby-cli\\\"\" did not complete successfully: exit code: 1" serviceContext="{bob }"
exit
exit
```

Any idea why? :thinking: 

Also, the build seems to be quite slow. Do we enforce the same cgroup limits to headless workspaces as we do for regular ones?

File: chart/templates/image-builder-configmap.yaml [hunk 4]
```diff
@@ -4,16 +4,6 @@
 {{ $comp := .Values.components.imageBuilder -}}
 {{- $this := dict "root" . "gp" $.Values "comp" $comp -}}
 
-{{- define "registry-name" -}}
```
Comment (geropl) [inline chart/templates/image-builder-configmap.yaml:7]: The function is still used below (cmp. [here](https://github.com/gitpod-io/gitpod/pull/4547/files#diff-f8e19dacea608343975006973dd477d4c4e0f27f3f84fbe551b2f9afef1ff917L33-L34)). If we remove here the fallbacks should go away as well.

**Comment by geropl** [comment]
I reviewed all code "around" the actually implementation and that looks good (see comments). Only issue I see is the [one mentioned above](https://github.com/gitpod-io/gitpod/pull/4547#issuecomment-884281490), else lgtm!

File: components/server/src/container-module.ts [hunk 5]
```diff
@@ -129,7 +129,7 @@ export const productionContainerModule = new ContainerModule((bind, unbind, isBo
     ).inSingletonScope();
 
     bind(ImageBuilderClientConfig).toDynamicValue(() => {
-        return { address: "image-builder:8080" }
+        return { address: process.env.IMAGE_BUILDER_SERVICE || "image-builder-mk3:8080" }
```
Comment (geropl) [inline components/server/src/container-module.ts:132]: This `process.env` call should really move to env, leaving us here with:
```
    bind(ImageBuilderClientConfig).toDynamicValue(ctx => {
        const env = ctx.container.get<Env>(Env);
        return { address: env.imageBuilderAddress }
    });
```

**Comment by csweichel** [comment]
@geropl thank you for looking into this. I've fixed the `procfd` issue by downgrading buildkit to 0.8.3 again. We don't support mounting proc through [`fsopen`](https://lwn.net/Articles/759499/) yet, only through `mount`.

File: components/server/src/container-module.ts (hunk 5 above)
Comment (csweichel) [reply components/server/src/container-module.ts]: done

**Comment by aledbf** [comment]
using internal docker registry (EKS) I see this error:
```
#11 ERROR: failed to do request: Head https://registry.default.svc.cluster.local/v2/workspace-images/blobs/sha256:f5cfc5f3861ca68ff5460aa1004a9e2c901c752efcf98e4d8fd89578b5689e0a: dial tcp: lookup registry.default.svc.cluster.local on 8.8.8.8:53: no such host
------
 > exporting to image:
------
FATA[0086] build failed                                  error="cannot build Gitpod layer: failed to solve: failed to do request: Head https://registry.default.svc.cluster.local/v2/workspace-images/blobs/sha256:f5cfc5f3861ca68ff5460aa1004a9e2c901c752efcf98e4d8fd89578b5689e0a: dial tcp: lookup registry.default.svc.cluster.local on 8.8.8.8:53: no such host" serviceContext="{bob }"
exit
exit
```

**Comment by csweichel** [comment]
> using internal docker registry (EKS) I see this error:
> 
> ```
> #11 ERROR: failed to do request: Head https://registry.default.svc.cluster.local/v2/workspace-images/blobs/sha256:f5cfc5f3861ca68ff5460aa1004a9e2c901c752efcf98e4d8fd89578b5689e0a: dial tcp: lookup registry.default.svc.cluster.local on 8.8.8.8:53: no such host
> ------
>  > exporting to image:
> ------
> FATA[0086] build failed                                  error="cannot build Gitpod layer: failed to solve: failed to do request: Head https://registry.default.svc.cluster.local/v2/workspace-images/blobs/sha256:f5cfc5f3861ca68ff5460aa1004a9e2c901c752efcf98e4d8fd89578b5689e0a: dial tcp: lookup registry.default.svc.cluster.local on 8.8.8.8:53: no such host" serviceContext="{bob }"
> exit
> exit
> ```

We used to run the registry through the proxy (prior to the caddy migration). I reckon that's the way forward here, too.

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 4f9bc2fd8cbb030b8863fa4a732b58c65ad8a22d</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/4547#issuecomment-884949137" title="Approved">aledbf</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/4547#" title="Author self-approved">csweichel</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **jankeromnes** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @jankeromnes` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)~~ [aledbf,csweichel]
- ~~[components/image-builder/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/image-builder/OWNERS)~~ [aledbf,csweichel]
- ~~[components/image-builder-api/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/image-builder-api/OWNERS)~~ [csweichel]
- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**
- ~~[components/ws-manager/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-manager/OWNERS)~~ [aledbf,csweichel]
- ~~[components/ws-manager-api/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-manager-api/OWNERS)~~ [aledbf,csweichel]
- ~~[components/ws-proxy/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-proxy/OWNERS)~~ [aledbf,csweichel]
- ~~[dev/gpctl/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/dev/gpctl/OWNERS)~~ [aledbf,csweichel]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by AlexTugarev** [comment]
Changes in compontents/server LGTM!

/approve

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/4547#issuecomment-884949137" title="Approved">aledbf</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/4547#issuecomment-884961080" title="Approved">AlexTugarev</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/4547#" title="Author self-approved">csweichel</a>*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)~~ [AlexTugarev,aledbf,csweichel]
- ~~[components/image-builder/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/image-builder/OWNERS)~~ [aledbf,csweichel]
- ~~[components/image-builder-api/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/image-builder-api/OWNERS)~~ [csweichel]
- ~~[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)~~ [AlexTugarev]
- ~~[components/ws-manager/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-manager/OWNERS)~~ [aledbf,csweichel]
- ~~[components/ws-manager-api/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-manager-api/OWNERS)~~ [aledbf,csweichel]
- ~~[components/ws-proxy/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-proxy/OWNERS)~~ [aledbf,csweichel]
- ~~[dev/gpctl/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/dev/gpctl/OWNERS)~~ [aledbf,csweichel]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**